sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubAuthenticationPolicy.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http.Headers;
+using System.Text;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Messaging.WebPubSub
+{
+    /// <summary>
+    /// Web PubSub Authentication Policy.
+    /// </summary>
+    internal partial class WebPubSubAuthenticationPolicy : HttpPipelineSynchronousPolicy
+    {
+        private readonly AzureKeyCredential _credential;
+
+        /// <summary>
+        /// Creates an instance of the authentication policy
+        /// </summary>
+        /// <param name="credential"></param>
+        public WebPubSubAuthenticationPolicy(AzureKeyCredential credential) => _credential = credential;
+
+        /// <inheritdoc/>
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            string audience = message.Request.Uri.ToUri().ToString();
```

Comment:
Always prefer `AbsoluteUri` to avoid encoding issues.
```suggestion
            string audience = message.Request.Uri.ToUri().AbsoluteUri;
```